connectors/source.py

```diff
@@ -105,20 +106,27 @@ def value(self):
     def value(self, value):
         self._value = value
 
-    def _convert(self, value, field_type):
-        if not isinstance(value, str):
-            # we won't convert the value if it's not a str
+    def _convert(self, value, field_type_):
+        cast_type = locate(field_type_)
+        if cast_type not in [str, int, float, bool, list]:
+            # unsupported type
             return value
 
-        if field_type == "int":
-            return int(value)
-        elif field_type == "float":
-            return float(value)
-        elif field_type == "bool":
-            return value.lower() in ("y", "yes", "true", "1")
-        elif field_type == "list":
-            return [item.strip() for item in value.split(",")]
-        return value
+        if isinstance(value, cast_type):  # pyright:ignore
+            return value
+
+        # list requires special type casting
+        if field_type_ == "list":
+            if isinstance(value, str):
+                return [item.strip() for item in value.split(",")] if value else []
+            else:
+                return [value] if value is not None else []
```

Comment:
Just curious, do you need to add special handling if `value` is `set` or `dict`?

Reply:
I added a check for if `value` is `set` or `dict` and the `field_type_` is list. They should be converted like this:

`{"foo": "bar"}` -> `[("foo", "bar")]`
`{"foo", "bar"}` -> `["foo", "bar"]`